app/src/main/java/net/gsantner/opoc/ui/FileSearchDialog.java

```diff
@@ -0,0 +1,224 @@
+            }
+        }
+    }
+
+    private static AlertDialog.Builder buildDialog(final Dialog initializer, final Callback.a1<CallBackOptions> dialogCallback) {
+        final AppSettings appSettings = new AppSettings(initializer._activity);
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(initializer._activity, appSettings.isDarkThemeEnabled() ? R.style.Theme_AppCompat_Dialog : R.style.Theme_AppCompat_Light_Dialog);
+
+        final ScrollView scrollView = new ScrollView(initializer._activity);
+        final LinearLayout dialogLayout = new LinearLayout(initializer._activity);
+        dialogLayout.setOrientation(LinearLayout.VERTICAL);
+        final int dp4px = (int) (new ContextUtils(dialogLayout.getContext()).convertDpToPx(4));
+        final int textColor = ContextCompat.getColor(initializer._activity, appSettings.isDarkThemeEnabled() ? R.color.dark__primary_text : R.color.light__primary_text);
+
+        final LinearLayout.LayoutParams margins = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+        margins.setMargins(dp4px * 5, dp4px, dp4px * 5, dp4px);
+
+        final LinearLayout.LayoutParams subCheckBoxMargins = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+        subCheckBoxMargins.setMargins(dp4px * 5 * 2, dp4px, dp4px * 5, dp4px);
+
+        final TextView messageTextView = new TextView(initializer._activity);
+        final AppCompatEditText searchEditText = new AppCompatEditText(initializer._activity);
+        Spinner queryHistorySpinner = new Spinner(initializer._activity);
+        final CheckBox regexCheckBox = new CheckBox(initializer._activity);
+        final CheckBox caseSensitivityCheckBox = new CheckBox(initializer._activity);
+        final CheckBox searchInContentCheckBox = new CheckBox(initializer._activity);
+        final CheckBox onlyFirstContentMatchCheckBox = new CheckBox(initializer._activity);
+
+        final Callback.a0 submit = () -> {
+            final String query = searchEditText.getText().toString();
+            if (dialogCallback != null && !TextUtils.isEmpty(query)) {
+                CallBackOptions callBackOptions = new CallBackOptions(query, regexCheckBox.isChecked(), caseSensitivityCheckBox.isChecked(), searchInContentCheckBox.isChecked(), onlyFirstContentMatchCheckBox.isChecked());
+                dialogCallback.callback(callBackOptions);
+            }
+        };
+
+        // TextView
+        messageTextView.setText(R.string.recursive_search_in_current_directory);
+        dialogLayout.addView(messageTextView, margins);
+
+        // EdiText: Search query input
+        searchEditText.setHint(R.string.search);
+        searchEditText.setSingleLine(true);
+        searchEditText.setMaxLines(1);
+        searchEditText.setTextColor(textColor);
+        searchEditText.setHintTextColor((textColor & 0x00FFFFFF) | 0x99000000);
```

Comment:
Should make the color a resource

Reply:
It is a color filter applied to a existing color. should be fine I'd say

Reply:
👍